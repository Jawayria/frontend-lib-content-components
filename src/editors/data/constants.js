// maybe put the string mapping in here?