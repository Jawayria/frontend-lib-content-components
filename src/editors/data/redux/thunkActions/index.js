import { StrictDict } from '../../../utils';

import app from './app';
import video from './video';

export default StrictDict({
  app,
  video,
});
