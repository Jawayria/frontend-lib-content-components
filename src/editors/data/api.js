import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
import { ActionStates, normalizeContent } from './constants';

async function getAsync(setValue, setError, setLoading, params) {
  try {
    setLoading(ActionStates.IN_PROGRESS);
    const result = await getAuthenticatedHttpClient().get(...params);
    setValue(result);
  } catch (e) {
    setError(e);
  } finally {
    setLoading(ActionStates.FINISHED);
  }
}

async function saveAsync(setInProgress, setResponse, params) {
  try {
    const result = await getAuthenticatedHttpClient().post(...params);
    setResponse(result);
  } catch (e) {
    setResponse(e);
  } finally {
    setInProgress(ActionStates.FINISHED);
  }
}

export function fetchBlockById(setValue, setError, setLoading, blockId, studioEndpointUrl) {
  const url = `${studioEndpointUrl}/xblock/${blockId}`;
  getAsync(setValue, setError, setLoading, [url]);
}

export function fetchUnitById(setValue, setError, setLoading, blockId, studioEndpointUrl) {
  const url = `${studioEndpointUrl}/xblock/${blockId}?fields=ancestorInfo`;
  getAsync(setValue, setError, setLoading, [url]);
}

export function saveBlock(blockId, blockType, courseId, studioEndpointUrl, content, setInProgress, setResponse) {
  const normalizedContent = normalizeContent(blockType, content, blockId, courseId);
  const url = `${studioEndpointUrl}/xblock/${encodeURI(blockId)}`;
  const params = [url, normalizedContent];
  saveAsync(setInProgress, setResponse, params);
}
