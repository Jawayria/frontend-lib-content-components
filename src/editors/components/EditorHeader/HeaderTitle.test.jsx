import React from 'react';
import { shallow } from 'enzyme';
import { selectors, thunkActions } from '../../data/redux';

import { EditableHeader } from './EditableHeader';

import { HeaderTitle } from './HeaderTitle';
import { thunkActions } from '../../data/redux';

jest.mock('./EditableHeader', () => 'EditableHeader');

describe('HeaderTitle', () => {
  let props = {
    editorRef: jest.fn().mockName('args.editorRef'),
    isInitialized: false,
    setBlockTitle: jest.fn().mockName('args.setBlockTitle'),
    typeHeader: 'html',
  };
  let el;
  beforeEach(() => {
    el = shallow(<HeaderTitle {...props} />);
  });
  describe('snapshots', () => {
    test('not initialized', () => {
      expect(shallow(<HeaderTitle {...props} />)).toMatchSnapshot();
    });
    test('initialized', () => {
      expect(shallow(<HeaderTitle {...props} isInitialized />)).toMatchSnapshot();
    });
  });

  describe('mapStateToProps');

  describe('mapDispatchToProps')
  
  
  // test('snapshot: initialized and not editing', () => {
  //   expect(shallow(<HeaderTitle {...props} isInitialized />)).toMatchSnapshot();
  // });
});