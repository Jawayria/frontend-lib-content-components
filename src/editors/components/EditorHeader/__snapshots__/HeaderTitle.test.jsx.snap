// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTitle snapshots initialized 1`] = `
<div
  className="d-flex"
>
  <div
    style={
      Object {
        "lineHeight": "1.5",
        "paddingRight": ".25em",
      }
    }
  >
    html
  </div>
  <IconButton
    alt="Edit"
    aria-label="Edit Title"
    className="mr-2"
    iconAs="Icon"
    onClick={[Function]}
    size="sm"
    src={[MockFunction icons.Edit]}
  />
</div>
`;

exports[`HeaderTitle snapshots not initialized 1`] = `
<FormattedMessage
  defaultMessage="Loading..."
  description="Message displayed while loading content"
  id="authoring.texteditor.title.loading"
/>
`;
